File: Patterns/Factory Method Pattern/Cpp/Pizza/NYStylePizza/NYStylePepperoniPizza.cpp
#include "../Pizza.cpp"

class NYStylePepperoniPizza : public Pizza {
public:
    NYStylePepperoniPizza() {
        name = "뉴욕 스타일 소스와 페페로니 피자";
        dough = "씬 크러스트 도우";
        sauce = "마리나라 소스";

        toppings.push_back("\t잘게 썬 페페로니\n");
    }
};

File: Patterns/Factory Method Pattern/CSharp/Pizza/NYStylePizza/NYStylePepperoniPizza.cs
using System;

namespace Pizza.NYStylePizza
{
    public class NYStylePepperoniPizza : Pizza 
    {
        public NYStylePepperoniPizza() :
            base("뉴욕 스타일 소스와 페페로니 피자", "씬 크러스트 도우", "마리나라 소스")
        {
            toppings.Add("\t잘게 썬 페페로니");
        }
    }
}